### ISSUE TITLE
First slide does not display properly on API 15 devices

### ISSUE THREAD

**Issue by drdla49** [PaoloRotolo/AppIntro#299]
I try to use appIntro in my Android project. 

On any other SDK then 15 everithing is OK. But on SDK 15 the first slide look as this  [screenshot](https://goo.gl/photos/MvgozEUWfxYTjWQD9)

Swipe not work, if I click on arrow to next slide, the next slide is OK, if I swipe back to first everithing is OK.
Second slide is [here](https://goo.gl/photos/uLKK3qRPtRe2Gkrx7)

Here is code of my IntroActivity:

`
public class WizardActivity extends AppIntro2 {
    private int mWizardId;
    private int mPosition;
    private List<Fragment> mSlides;
    private WizardConf mWizardConf = new WizardConf();

    public WizardActivity() {
    }

    public void init(@Nullable Bundle savedInstanceState) {
        Bundle bundle = this.getIntent().getExtras();
        this.mWizardId = bundle.getInt("wizardId");
        boolean slidesToShow = false;
        int[] screens = this.mWizardConf.getScreens(this.mWizardId);
        int numberOfPages = 0;
        boolean hasBlank = false;

        for(int window = 0; window < screens.length && (!SDKHelper.isPreMarshamallow() || screens[window] != 100); ++window) {
            if(screens[window] == 100) {
                hasBlank = true;
            }

            if(!WizardPrefs.wasWizardPageShown(this.mWizardId, screens[window]) || savedInstanceState != null) {
                slidesToShow = true;
                WizardScreen screen = this.mWizardConf.getScreen(screens[window]);
                this.addFragment(screen);
                ++numberOfPages;
                if(!SDKHelper.isPreMarshamallow() && screen.getRequestedPermission() != null && this.checkSelfPermission(screen.getRequestedPermission()) != 0) {
                    this.askForPermissions(new String[]{screen.getRequestedPermission()}, numberOfPages);
                }
            }
        }

        this.mSlides = this.getSlides();
        if(!slidesToShow || hasBlank && numberOfPages == 1) {
            Intent var9 = new Intent();
            this.setResult(-1, var9);
            this.finish();
        }

        this.setProgressButtonEnabled(true);
        this.setNavBarColor(this.getResources().getString(this.mWizardConf.getScreen(0).getBottomBackgroundColor()));
        this.getPager().addOnPageChangeListener(new OnPageChangeListener() {
            public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {
            }

            public void onPageSelected(int position) {
                WizardActivity.this.mPosition = position;
                WizardScreen screen = ((WizardBaseFragment)WizardActivity.this.mSlides.get(position)).getScreen();
                if(position > 0) {
                    WizardActivity.this.savePref(((WizardBaseFragment)WizardActivity.this.mSlides.get(position - 1)).getScreen());
                }

                if(screen.getId() == 100) {
                    if(WizardActivity.this.checkSelfPermission(((WizardBaseFragment)WizardActivity.this.mSlides.get(position - 1)).getScreen().getRequestedPermission()) == 0) {
                        WizardActivity.this.sendResult(-1);
                    } else {
                        WizardActivity.this.sendResult(0);
                    }
                }

                WizardActivity.this.setNavBarColor(WizardActivity.this.getResources().getString(WizardActivity.this.mWizardConf.getScreen(position).getBottomBackgroundColor()));
            }

            public void onPageScrollStateChanged(int state) {
            }
        });
        if(!SDKHelper.isPreLollipop()) {
            Window var10 = this.getWindow();
            var10.clearFlags(67108864);
            var10.addFlags(-2147483648);
            var10.setStatusBarColor(this.getResources().getColor(color.color_wizard_default_bottom_background));
        }

    }

    public void addFragment(WizardScreen screen) {
        WizardBaseFragment wizardFragment = screen.getFragmentInstance();
        this.addSlide(wizardFragment);
    }

    public void onNextPressed() {
    }

    public void onDonePressed() {
        this.savePref(((WizardBaseFragment)this.mSlides.get(this.mSlides.size() - 1)).getScreen());
        this.sendResult(-1);
    }

    public void sendResult(int result) {
        Intent returnIntent = new Intent();
        this.setResult(result, returnIntent);
        this.finish();
    }

    public void onSlideChanged() {
    }

    public void onBackPressed() {
    }

    private void savePref(WizardScreen screen) {
        if(screen != null && screen.getId() != 100) {
            if(!SDKHelper.isPreMarshamallow() && screen.getRequestedPermission() != null) {
                if(this.checkSelfPermission(screen.getRequestedPermission()) == 0) {
                    this.setPageShown(this.mWizardId, screen.getId(), true);
                } else {
                    this.setPageShown(this.mWizardId, screen.getId(), false);
                }

                WizardPrefs.setPermissionShown(screen.getRequestedPermission());
            } else {
                this.setPageShown(this.mWizardId, screen.getId(), true);
            }
        }

    }

    private void setPageShown(int wizardId, int screenId, boolean value) {
        WizardPrefs.setWizardPageShown(wizardId, screenId, value);
    }

    public void onRequestPermissionsResult(int requestCode, String[] permissions, int[] grantResults) {
        if(grantResults.length > 0 && ((WizardBaseFragment)this.mSlides.get(this.mPosition + 1)).getScreen().getId() == 100) {
            if(grantResults[0] == 0) {
                this.savePref(((WizardBaseFragment)this.mSlides.get(this.mPosition)).getScreen());
                this.sendResult(-1);
                GoogleAnalyticsHelper.sendEvent("Education", "PermissionAllow", permissions[0], Long.valueOf(0L));
            } else {
                this.sendResult(0);
            }
        }
    }}
`

Thank you for any answer.

**Comment by avluis** [comment]
Please follow the flipping format provided.
And show code.

**Comment by Andrew-Quebe** [comment]
@avluis I think we can close this since AppIntro now has wizard mode. Am I wrong?

**Comment by avluis** [comment]
I don't think the issue applied to Wizard Mode but rather that the first slide was not being displayed correctly on API 15 devices -- but after looking at the code that was edited in there, I'm not sure I'm even sure where to begin reading said code.

**Comment by Andrew-Quebe** [comment]
@drdla49 I think it's best for you to redo your code for all this using the new wizard mode when it's released. Your code is incredibly complex and hard to follow. Please reopen this issue if you have any objections, or have simplified your code.
